src/utils/subgraph.ts

```diff
@@ -0,0 +1,285 @@
+import { gql, OperationResult, TypedDocumentNode, OperationContext } from 'urql'
+import { DDO } from '@oceanprotocol/lib'
+import { getUrqlClientInstance } from '../providers/UrqlProvider'
+import { getOceanConfig } from './ocean'
+import {
+  AssetsPoolPrice,
+  AssetsPoolPrice_pools as AssetsPoolPricePool
+} from '../@types/apollo/AssetsPoolPrice'
+import {
+  AssetsFrePrice,
+  AssetsFrePrice_fixedRateExchanges as AssetsFrePriceFixedRateExchange
+} from '../@types/apollo/AssetsFrePrice'
+import {
+  AssetsFreePrice,
+  AssetsFreePrice_dispensers as AssetFreePriceDispenser
+} from '../@types/apollo/AssetsFreePrice'
+import { BestPrice } from '../models/BestPrice'
+
+export interface UserLiquidity {
+  price: string
+  oceanBalance: string
+}
+
+export interface PriceList {
+  [key: string]: string
+}
+
+export interface AssetListPrices {
+  ddo: DDO
+  price: BestPrice
+}
+
+interface DidAndDatatokenMap {
+  [name: string]: string
+}
```

Comment:
suggestion(blocking): lets move the types/interfaces into the `src/@types` directory